models/facial_expression_recognition/demo.py

```diff
@@ -0,0 +1,163 @@
+        (255,  0,   0),  # right eye
+        (0,    0, 255),  # left eye
+        (0,  255,   0),  # nose tip
+        (255,  0, 255),  # right mouth corner
+        (0,  255, 255)   # left mouth corner
+    ]
+
+    for det, fer_type in zip(results, fer_res):
+        bbox = det[0:4].astype(np.int32)
+        cv.rectangle(output, (bbox[0], bbox[1]), (bbox[0]+bbox[2], bbox[1]+bbox[3]), box_color, 2)
+
+        cv.putText(output, FacialExpressionRecog.getDesc(fer_type), (bbox[0], bbox[1]+12), cv.FONT_HERSHEY_DUPLEX, 0.5, text_color)
+
+        landmarks = det[4:14].astype(np.int32).reshape((5, 2))
+        for idx, landmark in enumerate(landmarks):
+            cv.circle(output, landmark, 2, landmark_color[idx], 2)
+
+    return output
+
+
+def process(detect_model, align_model, fer_model, frame):
+    dets = np.zeros((0, 15))
+    for scale in [0.25, 0.5, 1.]:
+        try:
+            resized_frame = cv.resize(frame, None, None, fx=scale, fy=scale, interpolation=cv.INTER_LINEAR)
+            h, w, _ = resized_frame.shape
+            detect_model.setInputSize([w, h])
+            det_result = detect_model.infer(resized_frame)
+            if det_result is None:
+                continue
+            det_result[:, :-1] /= scale
+            dets = np.concatenate((dets, det_result), axis=0)
+        except Exception:
+            pass
+
+    _boxes = copy.deepcopy(dets[:, :4])
+    keepIdx = cv.dnn.NMSBoxes(
+        bboxes=_boxes.tolist(),
+        scores=dets[:, -1].tolist(),
+        score_threshold=args.conf_threshold,
+        nms_threshold=args.nms_threshold,
+        top_k=args.top_k,
+    )
+
+    fer_res = np.zeros(0, dtype=np.int8)
+    if len(keepIdx) > 0:
+        dets = dets[keepIdx]
+        for face_points in dets:
+            img = align_model.get_align_image(frame, face_points[4:-1].reshape(-1, 2))
+            fer_res = np.concatenate((fer_res, fer_model.infer(img)), axis=0)
+
+    return dets, fer_res
+
+
+if __name__ == '__main__':
+    detect_model = YuNet(modelPath=args.detect_model,
+                         inputSize=[320, 320],
+                         confThreshold=args.conf_threshold,
+                         nmsThreshold=args.nms_threshold,
+                         topK=args.top_k,
+                         backendId=args.backend,
+                         targetId=args.target)
+
+    align_model = FaceAlignment()
+
+    fer_model = FacialExpressionRecog(modelPath=args.fer_model,
+                                      backendId=args.backend,
+                                      targetId=args.target)
+
+    # If input is an image
+    if args.input is not None:
+        image = cv.imread(args.input)
+
+        # Get detection and fer results
+        dets, fer_res = process(detect_model, align_model, fer_model, image)
+
+        # Draw results on the input image
+        image = visualize(image, dets, fer_res)
+
+        print('%s %3d faces detected.' % (datetime.datetime.now(), len(dets)))
```

Comment:
Some times the text on face bbox is not clear enough to recognize. How about moving the print into `visualize` and  print faces with coordinates and facial expression?